Library/Homebrew/cmd/tap-info.rb

```diff
@@ -48,7 +48,7 @@ def print_tap_info(taps)
       formula_count = 0
       command_count = 0
       private_count = 0
-      Tap.each do |tap|
+      Tap.select(&:installed?).each do |tap|
```

Comment:
Why do these all need changed? Does that mean `Tap.each` will no longer work as expected in usage in taps/scripts?

Reply:
It will now also return `CoreTap` and `CaskCoreTap` when using the API, even if these are not installed locally.

This is an undocumented API in any case. The only two uses I could find so far are [in `brew bundle`](https://github.com/Homebrew/homebrew-bundle/blob/d3cf1e126dc4686e75c9b3013caf08558aba241c/lib/bundle/tap_dumper.rb#L37), which only checks the `Tap#remote `, which is also respected when using the API, and [in `brew which-update`](https://github.com/Homebrew/homebrew-command-not-found/blob/ae70e2c3db04971a93af8397e860d28ff98364b8/lib/executables_db.rb#L61), which is only run in CI without the API.

Reply:
Why does the API need to change here?

> It will now also return `CoreTap` and `CaskCoreTap` when using the API, even if these are not installed locally.

`Tap.each` will do that or `Tap.select(&:installed?)` or both or neither?

Reply:
@reitermarkus Thanks for explanation, I understand now.